src/main/relation/join_relation.cpp

```diff
@@ -35,6 +39,121 @@ unique_ptr<QueryNode> JoinRelation::GetQueryNode() {
 	return std::move(result);
 }
 
+namespace {
+
+class BindingDeduplicator {
+public:
+	BindingDeduplicator() {
+	}
+
+public:
+	unique_ptr<TableRef> DeduplicateBindings(unique_ptr<TableRef> original) {
+		auto &ref = *original;
+		switch (ref.type) {
+		case TableReferenceType::BASE_TABLE:
```

Comment:
In order to simplify this PR, can we maybe remove this and get the simpler fixes in first? Then we can revisit this in the future.